Utilities/CredentialUtils.cs

```diff
@@ -75,37 +75,39 @@ public static bool HasBadStartOrEndChar(string credentialValue)
         /// </summary>
         /// <param name="vcapServices">Dictionary<string, List<VcapCredential>> representing the VCAP_SERVICES</param>
         /// <param name="serviceName">The name of the service whose credentials are sought</param>
-        /// <param name="plan">The name of the plan for which the credentials are sought, e.g. 'standard', 'beta' etc, may be null</param>
         /// <returns>The first set of credentials that match the search criteria, service name and plan. May return null</returns>
-        private static VcapCredential GetVcapCredentialsObject(Dictionary<string, List<VcapCredential>> vcapServices, string serviceName, string plan = null)
+        private static VcapCredential GetVcapCredentialsObject(Dictionary<string, List<VcapCredential>> vcapServices, string serviceName)
         {
-            if (vcapServices == null || vcapServices.Count == 0)
-            {
-                return null;
-            }
-
-            foreach (KeyValuePair<string, List<VcapCredential>> kvp in vcapServices)
-            {
-                if (kvp.Key.StartsWith(serviceName))
+            if (vcapServices == null || vcapServices.Count == 0)
+            {
+                return null;
+            }
+            // search the inner entries of the vcap for the matching serviceName
+            foreach (KeyValuePair<string, List<VcapCredential>> kvp in vcapServices)
+            {

```

Comment:
There are a lot of whitespace changes in this file.  I don't know if these are appropriate or not.

Reply:
@mkistler - I've made some updates that I think helped. Let me know what you think.